### PR TITLE
Document assertSameAsFile

### PR DESCRIPTION
**Pull request by AD7six**
Mostly applicable to bake tests in the core

Ref https://github.com/cakephp/cakephp/pull/8169

### CODE REVIEW

File: en/development/testing.rst [hunk 1]
```diff
@@ -1011,6 +1011,62 @@ assertions in `TestSuite
...
+        public function setUp()
+        {
+            $this->_compareBasePath = APP . 'tests/comparisons/';
+            parent::setUp();
+        }
+
+        public function testExample()
+        {
+            $result = ...;
+            $this->assertSameAsFile('example.php', $result);
```
Comment (ravage84) [inline en/development/testing.rst]: Just curious: Are absolute paths possible? Even if ``$this->_compareBasePat`` is set? If so, what about Windows?

Comment (AD7six) [inline en/development/testing.rst]: Not 100% sure what you mean - `assertSameAsFile` expects only a filename.

I'll update the example to use the DS constant for compareBasePath, as [existing usage of this assert method does](https://github.com/cakephp/bake/blob/20c48a87150b50063ab0e8bec46436edd401c45e/tests/TestCase/View/BakeViewTest.php#L42).

Comment (ravage84) [inline en/development/testing.rst]: I mean if I have one file that I want to compare with that isn't in the ``$this->_compareBasePath``. I would want to provide a absolute path for the ``example.php`.

Comment (HavokInspiration) [inline en/development/testing.rst]: No, the ``_compareBasePath`` property is always put before the filename given.
You'd have to temporarily set it to an empty string if you want to provide a full path as a filename.

See : https://github.com/cakephp/cakephp/blob/master/src/TestSuite/StringCompareTrait.php#L44

**Comment by ravage84** [comment]
Excactly, I do not expect such behavior if I give an absolute path.
To say it in a different way, when I provide an absolute path, I expect the function to respect that... Just sayin'.

**Comment by AD7six** [comment]
> I expect the function to respect that

Heard, but I'm not changing it =)

File: en/development/testing.rst (hunk 1 above)
Comment (dereuromark) [inline en/development/testing.rst]: I think the "what about Windows" was mainly about the different PHP_EOL chars.
And that those should maybe have to be unified before comparison (not sure of the assertSameAsFile() wrapper already does that).

Comment (AD7six) [inline en/development/testing.rst]: assertSameAsFile is not comparing the path to files - it's comparing the contents @dereuromark

Comment (dereuromark) [inline en/development/testing.rst]: My note was about the content ;) Content is stored usually as per OS PHP_EOL, thus comparing two contents without the normalization of this char can lead to fails on different OS.

**Pull request merged**